base/loading.jl

```diff
@@ -1241,6 +1250,10 @@ function compilecache(pkg::PkgId, path::String)
     end
 end
 
+const PRECOMPILING_INTERACTIVELY = Ref{Bool}(false)
+precompiling_interactively!(b::Bool) = PRECOMPILING_INTERACTIVELY[] = b
```

Comment:
Imo just use the global variable. Makes it more clear from looking at the code what is happening.